docs/sriov.md

```diff
@@ -0,0 +1,210 @@
+```
+
+Also, make sure that vfio devices are accessible to kvm group by configuring udev rules:
+
+```
+$ cat /etc/udev/rules.d/10-qemu-hw-users.rules
+SUBSYSTEM=="vfio", OWNER="root", GROUP="root", MODE="0666"
+KERNEL=="vfio", SUBSYSTEM=="misc", OWNER="root", GROUP="root", MODE="0666"
+KERNEL=="kvm", GROUP="root", MODE="0666"
+$ udevadm control --reload-rules && udevadm trigger
+```
+
+Now you are ready to set up your cluster.
+
+# Set up kubernetes cluster
+
+You can use your preferred mechanism to deploy your kubernetes cluster as long
+as you deploy on bare metal.
+
+In the following example, we configure the cluster using `local` provider which
+is part of kubevirt/kubevirt repo. Please consult cluster/local/README.md for
+general information on setting up a host using the `local` provider.
+
+First, install default CNI plugins:
+
+```
+$ go get -u -d github.com/containernetworking/plugins/
+$ cd $GOPATH/src/github.com/containernetworking/plugins/
+$ ./build.sh
+$ mkdir -p /opt/cni/bin/
+$ cp bin/* /opt/cni/bin/
+```
+
+Then, prepare kubernetes tree for CNI enabled deployment:
+
+```
+$ go get -u -d k8s.io/kubernetes
+$ cd $GOPATH/src/k8s.io/kubernetes
+$ git diff
+diff --git a/hack/local-up-cluster.sh b/hack/local-up-cluster.sh
+index bcf988b..9911eed 100755
+--- a/hack/local-up-cluster.sh
++++ b/hack/local-up-cluster.sh
+@@ -639,6 +639,8 @@ function start_controller_manager {
+       --use-service-account-credentials \
+       --controllers="${KUBE_CONTROLLERS}" \
+       --leader-elect=false \
++      --cert-dir="$CERT_DIR" \
++      --allocate-node-cidrs=true --cluster-cidr=10.244.0.0/16 \
+       --master="https://${API_HOST}:${API_SECURE_PORT}" >"${CTLRMGR_LOG}" 2>&1 &
+     CTLRMGR_PID=$!
+ }
+export NET_PLUGIN=cni
+export CNI_CONF_DIR=/etc/cni/net.d/
+export CNI_BIN_DIR=/opt/cni/bin/
+```
+
+Install etcd:
+
+```
+$ ./hack/install-etcd.sh
+```
+
+Use `local` provider for kubevirt:
+
+```
+$ export KUBEVIRT_PROVIDER=local
+```
+
+Now finally, deploy kubernetes:
+
+```
+$ cd $GOPATH/src/kubevirt.io/kubevirt
+$ make cluster-up
+```
+
+Once the cluster is deployed, we can move to SR-IOV specific components.
+
+# Deploy SR-IOV services
+
+First, deploy Multus with default Flannel backend:
+
+```
+$ go get -u -d github.com/intel/multus-cni
+$ cd $GOPATH/src/github.com/intel/multus-cni/
+$ docker build .
+$ vi images/multus-daemonset.yml # change to refer to local multus:latest
+$ mkdir -p /etc/cni/net.d
+$ cp images/70-multus.conf /etc/cni/net.d/
+$ ./cluster/kubectl.sh create -f $GOPATH/src/github.com/intel/multus-cni/images/multus-daemonset.yml
+$ ./cluster/kubectl.sh create -f $GOPATH/src/github.com/intel/multus-cni/images/flannel-daemonset.yml
+```
+
+Now, deploy SR-IOV device plugin.
+
+```
+$ go get -u -d github.com/intel/sriov-network-device-plugin/
+$ cd $GOPATH/src/github.com/intel/sriov-network-device-plugin/images
+$ ./build_docker.sh
```

Comment:
Again, do we need to build it? It sounds inconvenient for multi-node deployments. Is this usage or development guide?

Reply:
It's development guide. There was an upload to their registry 3 days ago, perhaps it contains the needed patches, I will check.